### ISSUE TITLE
"meta" widget is not rendering

### ISSUE THREAD

**Issue by maneesha** [carpentries/carpentries-hugo-theme#12]
In the [sidebar layout](https://github.com/carpentries/carpentries-hugo-theme/blob/main/layouts/_default/sidebar.html), widgets for `feedback`, `meta`, and `edit-github` should display (with others defined in the page's frontmatter).  The meta widget is not rendering, resulting in two lines with an empty space in between them in the sidebar.

**Comment by maneesha** [comment]
I am still seeing the blank space between two lines.  See for example [this page](https://lc-new-website.netlify.app/lessons/) built from [this file](https://github.com/LibraryCarpentry/lc-new-website/blob/main/content/lessons.md).

**Comment by maneesha** [comment]
See screenshot:
![Screenshot from 2024-08-09 11-44-03](https://github.com/user-attachments/assets/c9c40144-dab1-4f87-b932-9a1a7165f903)

**Comment by maneesha** [comment]
Removing "Ready for Review" label as this feature is not yet working.

**Comment by AdamChlan** [comment]
ready for review

**Comment by maneesha** [comment]
@AdamChlan I am still seeing the black space (just like the screenshot above).

**Comment by maneesha** [comment]
Also, I noticed that before these changes, some widgets were included on all pages. [See the layout here](https://github.com/carpentries/carpentries-hugo-theme/blob/adamchlan/client-feedback/layouts/_default/sidebar.html).

Now, only explicitly specified widgets are included.  Can we have certain widgets appear on all pages? I don't remember discussing this change.

**Comment by AdamChlan** [comment]
yes, I can revert that change

**Comment by AdamChlan** [comment]
This has been reverted, and believe I have the rules working as expected

**Comment by AdamChlan** [comment]
Also for meta to render you need to add a `lastmod` parameter with  to the front matter like this:

`lastmod: 2022-12-11T14:45:08+00:00`

**Comment by maneesha** [comment]
OK,thanks.  That is what had been missing -  I thought the "last modified" data was automatically updated from the file history.  Is that possible to do?

**Comment by AdamChlan** [comment]
@maneesha yes, this has been adjusted.

**Comment by maneesha** [comment]
@AdamChlan  by "this has been adjusted" did you mean that it should now automatically use the file's last modified date/time?  I am only seeing the meta  widget render if the `lastmod` parameter is explicitly set.

**Comment by maneesha** [comment]
Also, in the [source for the meta widget](https://github.com/carpentries/carpentries-hugo-theme/blob/main/layouts/partials/widgets/meta.html), it looks like if the date format is not set, it should default to the format `2 April, 2006`.  

If I don't have that param set, I am still getting a long date (like `Last edited: 15 April, 2023 at 00:00:00 UTC`).  It is also not reading the date correctly.  The day and year are interpreted correctly but the month always says April no matter what month I put in.  [See example in this PR](https://github.com/datacarpentry/new-website/pull/14).

**Comment by regisphilibert** [comment]
@AdamChlan ref month should be `January`

```diff
+ Last edited: {{ dateFormat (site.Params.date_format_with_time | default "2 January, 2006") . }}
- Last edited: {{ dateFormat (site.Params.date_format_with_time | default "2 April, 2006") . }}
```

**Comment by AdamChlan** [comment]
Date format has been updated.

**Comment by AdamChlan** [comment]
This has been updated

**Comment by maneesha** [comment]
It looks like the issue with the date always being April has been fixed.

Also want to check about my question above - does the date have to be manually set in the frontmatter  or can it be automatically updated from the file history?

**Comment by AdamChlan** [comment]
@maneesha To have the modified dates pull in from Github, add the following to your projects hugo.yaml file:

```enableGitInfo: true```

This will completely automate the process so you will not need to add lastmod to the frontmatter

**Comment by maneesha** [comment]
Setting the `enableGitInfo` param seems to work.
I would like this to show just date, not date-time so I'v put in PR #39.

**Comment by maneesha** [comment]
This can be closed once #39 is reviewed/merged.
